File: gunkim/leetcode/array/121.Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;

        for(int i = 0; i < prices.length; i++) {
            if(prices[i] < min) {
                min = prices[i];
            } else if(prices[i] - min > max) {
                max = prices[i] - min;
            }
        }
        return max;
    }
}
class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.maxProfit(new int[]{7,1,5,3,6,4}));
    }
}

File: 100milliongold/leetcode/array/121.Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let price_min = 10000 // 최저 주식값
    let profit = 0 // 수익
    for(let price_current in prices){
        // 현재 가격 - 최저값의 값과 수익의 값 중, 최대값을 수익값으로 보고 스왑함
        price_min = Math.min(price_current, price_min)
            
        //현재 가격 - 최저값의 값과 수익의 값 중, 최대값을 수익값으로 보고 스왑함
        profit = Math.max(profit, price_current-price_min) 
    }
    return profit
};
